core/serialization/procedures.ts

```diff
@@ -98,14 +87,10 @@ export function loadProcedure<
 >(
   procedureModelClass: ProcedureModelConstructor<ProcedureModel>,
   parameterModelClass: ParameterModelConstructor<ParameterModel>,
-  state: State,
+  state: ProcedureState,
   workspace: Workspace,
 ): ProcedureModel {
-  const proc = new procedureModelClass(
-    workspace,
-    state.name,
-    state.id,
-  ).setReturnTypes(state.returnTypes);
+  const proc = procedureModelClass.loadState(state, workspace);
```

Comment:
I think it makes more sense to have it here, because if you have it be handled by the procedure model, your models become tightly coupled. You can't modify the serialization of your parameter model without modifying how your procedure model deserializes it. (Even if the modification is as simple as calling the new static method)

I can remove the internal methods though! I didn't notice they were internal otherwise I would have just done that. Thanks for the catch.

Reply:
Discussed offline. The issue isn't with `saveState`, the issue is with `loadState`! Because the parameter model's `loadState` is static, you have to make sure your procedure model knows which class to call `loadState` on. This means that if you create a parameter model subclass with a custom `loadState` + `saveState`, you have to modify your procedure model as well.

If they were all instance methods, this wouldn't matter. And I agree it would be better to have the procedure model handle all of the serialization.

But because `loadState` makes more sense as a static method, I think the current setup is the best compromise for making the models as extensible as possible (no tight coupling).